ChronoAPI/ChronoLog/include/RPCVisor.h

```diff
@@ -60,7 +62,11 @@ class RPCVisor {
             LOGE("client id is invalid");
             return CL_ERR_INVALID_ARG;
         }
-        return g_clientRegistryManager->remove_client_record(client_id, flags);
+        if (g_clientRegistryManager) {
+            return g_clientRegistryManager->remove_client_record(client_id, flags);
+        } else {
+            return CL_ERR_UNKNOWN;
+        }
     }
```

Comment:
ClientRegistry, MetadataMaps, etc should be instantiated before any of the RPC communication objects are, no need to check validity of the pointer on every client API call. If we structure our code correctly g_clientRegistryManager == nullptr case would be prevented by the compiler.  Either ClientRegistry::GetInstance() should be called in RPCVisor constructor or the reference to a valid ClientRegistry instance should be passed to RPCVisor constructor.   I think Aparna already started making GetInstance changes on a different branch...